core/platform/lf_arduino_support.c

```diff
@@ -0,0 +1,84 @@
+/* Arduino Platform API support for the C target of Lingua Franca. */
+
+/*************
+Copyright (c) 2021, The University of California at Berkeley.
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+
+/** Arduino API support for the C target of Lingua Franca.
+ *  
+ *  @author{Anirudh Rengarajan <arengarajan@berkeley.edu>}
+ */
+
+
+#include <time.h>
+#include <errno.h>
+
+#include "lf_arduino_support.h"
+#include "../platform.h"
+#include "Arduino.h"
+
+instant_t ns_to_microsec(instant_t time) {
+    return (time / 1000) - ((time / 1000) % 4);
+}
+
+/**
+ * Pause execution for a number of nanoseconds.
+ *
+ * @return 0 for success, or -1 for failure. In case of failure, errno will be
+ *  set appropriately (see `man 2 clock_nanosleep`).
+ */
+int lf_nanosleep(instant_t requested_time) {
+    unsigned int microsec = (unsigned int) requested_time;
+    if(microsec < 3) {
+        Serial.print("Cannot set delay under 3 microseconds due to precision problems");
+        return -1;
+    }
+    else if(microsec <= 16383) {
+        delayMicroseconds(microsec);
+    }
+    else {
+        delay(microsec / 1000);
+    }
+    return 0;
+}
+
+/**
+ * Initialize the LF clock.
+ */
+void lf_initialize_clock() {}
+
+/**
+ * Fetch the value of _LF_CLOCK (see lf_linux_support.h) and store it in tp. The
+ * timestamp value in 't' will always be epoch time, which is the number of
+ * nanoseconds since January 1st, 1970.
```

Comment:
This comment is no longer correct, right? First, it should be microseconds, not nanoseconds.  Second, I doubt it's epoch time because the number of microseconds since 1970 won't fit in 32 bits.

Reply:
Fixed. Had comments copied over from other support files but forgot to change them!